CMake/Version.cmake

```diff
@@ -0,0 +1,72 @@
+
+# Try to figure out  versioning scheme for dynamic libraries
+# based on the system we're compiling for
+function(determine_version_type _out_version_type_name)
+
+        set(_ver_type "none")
+
+        if(WIN32 OR CYGWIN)
+                set(_ver_type "windows")
+        else()
+                # this part tries to mimic autotools as close as possible
+                if(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Linux" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR
```

Comment:
So what about "normal" FreeBSD?

Reply:
In autotools there is detection of FreeBSD-a.out and FreeBSD-elf. I 'just' have to code it.